Snyk/resolve-deps
# feat: unique deps

- [x] Reviewed by @edwardgeorge 
#### What's this PR do?

Returns a faked package object that has only the unique dependencies (from the original method call).

This adds api `res.unique() => unique` and CLI arg `snyk-resolve -u`

A unique dependency is name@version, not just the package name.
#### Where should the reviewer start?

Primary logic is in `lib/unique.js`
#### How should this be manually tested?

Clone repo, in the dir `npm link`, then run `snyk-resolve -u` in the same directory. You'll see a unique list of packages (compared to `snyk-resolve` which is the complete tree, dupes included).
#### What are the relevant tickets?

https://snyksec.atlassian.net/browse/SC-879 - though this is just the start of the logic to get unique count.
